java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -263,8 +254,13 @@ private void cancel() {
             return;
         }
 
-        resultTable.handleBarrageError(new RequestCancelledException("Barrage subscription cancelled by client"));
-        GrpcUtil.safelyCancel(observer, "Barrage subscription is cancelled", null);
+        final RequestCancelledException cancelledException =
+                new RequestCancelledException("Barrage subscription cancelled by client");
+        if (!isSnapshot) {
+            // Stop our result table from processing any more data.
+            resultTable.forceReferenceCountToZero();
```

Comment:
We said that `BarrageTable.destroy` should call `cleanup()`, and that the assertion from cleanup should move to `realRefresh`.